msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/AadInstanceDiscoveryProvider.java

```diff
@@ -67,10 +67,9 @@ static InstanceDiscoveryMetadataEntry getMetadataEntry(URL authorityUrl,
 
             //If region autodetection is enabled and a specific region not already set,
             // set the application's region to the discovered region so that future requests can skip the IMDS endpoint call
-            if (msalRequest.application().azureRegion() == null && msalRequest.application().autoDetectRegion()) {
-                if (detectedRegion != null) {
+            if (msalRequest.application().azureRegion() == null && msalRequest.application().autoDetectRegion()
+                        && null != detectedRegion) {
```

Comment:
nit: keep constant on the right side? https://stackoverflow.com/questions/2369226/object-null-or-null-object :)